daemon/src/position_metrics.rs

```diff
@@ -0,0 +1,344 @@
+use crate::db;
+use async_trait::async_trait;
+use futures::StreamExt;
+use model::CfdEvent;
+use model::EventKind;
+use model::OrderId;
+use model::Position;
+use model::Usd;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::Decimal;
+use std::time::Duration;
+use tokio_tasks::Tasks;
+use xtra_productivity::xtra_productivity;
+use xtras::SendInterval;
+
+// TODO: ideally this would be more often
+pub const UPDATE_METRIC_INTERVAL_SECONDS: Duration = Duration::from_secs(60);
```

Comment:
Similar to projection, you could ping this actor a message every time an event gets appended to a particular CFD and then update the metrics accordingly.

Reply:
eventually, it would be nice if the actors that are concerned about this situation could subscribe to the process manager themselves and got notified when it happens (so that process manager doesn't need to be aware of projection / metrics / whatever, just dispatch a message that could be handled by different actors. 
Some sort of Observer pattern, but with actors :)

Reply:
Other than aesthetics, is there a need to decouple the process manager from this knowledge? Making the observation dynamic also has downsides: It is harder to understand, who gets notified when. At least at the moment, one can trace the source code to see where messages land.

If the process manager is merely iterating through a list of `MessageChannel<CfdChanged>`, this won't be as easy.

Reply:
this would decouple the software components - low-level layers should not know about the high-level ones.
in my opinion it would make things less error-prone and more modular.
As for understanding - is it hard to understand or follow code for classic OO observers? it's quite easy in an IDE to look at usages, in this case, which actors handle the given message.

Reply:
> this would decouple the software components - low-level layers should not know about the high-level ones.

I agree with that in general but it is also important to remember that decoupling per se isn't the end goal! Decoupling is for example useful if two components want/need to change at a different pace. Both, process manager and projection primarily process events meaning they are already semantically coupled.

> As for understanding - is it hard to understand or follow code for classic OO observers?

It is certainly hard_er_ than components which are wired together explicitly.

I am not necessarily opposed to the idea of OO-style observers. I am simply pointing out that if the nature of the problem does not require dynamic subscribing and unsubscribing, we are introducing a bit more complexity than fundamentally needed. In other words, there could be a simpler solution and not going for that one might confuse future readers.

There is a quote I like which says: 
["All problems in computer science can be solved by another level of indirection."](https://en.wikipedia.org/wiki/Indirection#cite_note-1)

I think we currently don't have a problem with the process manager knowing about projection so the cost is hard to justify IMO.